webapp/packages/core-theming/src/ThemeService.ts

```diff
@@ -81,13 +91,14 @@ export class ThemeService extends Bootstrap {
     });
   }
 
-  register(): void { }
+  register(): void {
+    this.loadAllThemes();
+  }
 
   async load(): Promise<void> {
-    this.loadAllThemes();
     this.setCurrentThemeId(this.defaultThemeId);
     this.settingsService.registerSettings(this.settings, THEME_SETTINGS_KEY);
-    await this.changeThemeAsync(this.currentThemeId);
+    await this.tryChangeTheme(this.userInfoResource.getConfigurationParameter(THEME_KEY) || this.currentThemeId);
```

Comment:
probably it always will be undefined (`this.userInfoResource.getConfigurationParameter(THEME_KEY)`), because user loaded after `load` phase in ui components (you can preload user info before `this.userInfoResource.load()`)